tests/test_odilo.py

```diff
@@ -0,0 +1,250 @@
+# encoding: utf-8
+import os
+import json
+import unittest
+
+from flask import (
+    Flask,
+)
+
+from api.odilo import (
+    OdiloAPI,
+    MockOdiloAPI,
+    OdiloCirculationMonitor,
+    FullOdiloCollectionMonitor,
+    RecentOdiloCollectionMonitor,
+)
+
+from api.circulation import (
+    CirculationAPI,
+    CirculationException
+)
+
+from sqlalchemy.ext.declarative import declarative_base
+from flask_sqlalchemy_session import flask_scoped_session
+from api.config import Configuration
+
+from . import (
+    DatabaseTest,
+    sample_data
+)
+
+from core.model import (
+    Collection,
+    ConfigurationSetting,
+    DataSource,
+    DeliveryMechanism,
+    ExternalIntegration,
+    Identifier,
+    Library,
+    LicensePool,
+    Representation,
+    RightsStatus,
+    SessionManager
+)
+
+Base = declarative_base()
+app = Flask(__name__)
+
+testing = True
+# testing = 'TESTING' in os.environ
+db_url = Configuration.database_url(testing)
+# Initialize a new database session unless we were told not to
+# (e.g. because a script already initialized it).
+autoinitialize = os.environ.get('AUTOINITIALIZE') != 'False'
+if autoinitialize:
+    SessionManager.initialize(db_url)
+session_factory = SessionManager.sessionmaker(db_url)
+_db = flask_scoped_session(session_factory, app)
+if autoinitialize:
+    SessionManager.initialize_data(_db)
+
+
+class OdiloAPITest(DatabaseTest, unittest.TestCase):
+    def setup(self):
+        super(OdiloAPITest, self).setup()
+        library = self._default_library
+        self._db = _db
+        self.collection = MockOdiloAPI.mock_collection(self._db)
+        self.circulation = CirculationAPI(self._db, library, api_map={ExternalIntegration.ODILO: MockOdiloAPI})
+        self.api = self.circulation.api_for_collection[self.collection.id]
+
+    def setUp(self):
+        self.setUp_common()
+        self._db = _db
+
+        # Load from DB or fake collection
+        self.collection = None
+        self.fake_collection()
+        self.api = OdiloAPI(self._db, self.collection)
+
+    def fake_collection(self):
```

Comment:
This seems redundant given that you create a mock_collection in the `setup` method.